### DISCUSSION TITLE
RabbitMQ super stream updation

### DISCUSSION THREAD

**Discussion by praveengopalan** [rabbitmq/rabbitmq-server#10684] [Questions]
Hi Team,
We have one requirement, update an existing super stream with a new partition. Now we added a new queue of type stream and binds that to the exchange (super stream). I just want to confirm whether it is a right approch?  if no, anyone please explain how I can update the existing super-stream by adding new partition.

**Comment by acogoluegnes** [comment]
This is the correct approach. Make sure to add an `x-stream-partition-order` argument to the binding with the appropriate value (see the existing bindings). This allows having a consistent order for partitions.

Restart  client applications, client libraries need to find out about the new partition.

Be aware that adding a new partition may compromise message order guarantee: if you use a given field to route messages to a partition (e.g. you publish invoices and the field is the customer ID), messages for a given field value are likely to end up in a different partition after the new partition addition.

**Comment by michaelklishin** [comment]
We'd appreciate a contribution to [the docs](https://github.com/rabbitmq/rabbitmq-website). Very likely are getting the software for free and some amount of support from the core team for free, so why not contribute a little bit back?
